<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from iiwa14.urdf.xacro              | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="iiwa">
  <!--joint between world and link_0-->
  <link name="world"/>
  <joint name="world_joint" type="fixed">
    <insert_block name="origin"/>
    <parent link="world"/>
    <child link="link_0"/>
  </joint>
  <link name="link_0">
    <inertial>
      <origin rpy="0 0 0" xyz="0 -0.01439 0.102503"/>
      <mass value="6.495"/>
      <inertia ixx="0.0690761441316632" ixy="0" ixz="0" iyy="0.0711085586914514" iyz="0" izz="0.0200625854402119"/>
    </inertial>
  </link>
  <!-- joint between link_0 and link_1 -->
  <joint name="joint_1" type="revolute">
    <parent link="link_0"/>
    <child link="link_1"/>
    <origin rpy="0 0 0" xyz="0 0 0.1525"/>
    <axis xyz="0 0 1"/>
    <limit effort="320" lower="-2.9670597283903604" upper="2.9670597283903604" velocity="10"/>
  </joint>
  <link name="link_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0 -0.01439 0.102503"/>
      <mass value="6.495"/>
      <inertia ixx="0.0690761441316632" ixy="0" ixz="0" iyy="0.0711085586914514" iyz="0" izz="0.0200625854402119"/>
    </inertial>
  </link>
  <!-- joint between link_1 and link_2 -->
  <joint name="joint_2" type="revolute">
    <parent link="link_1"/>
    <child link="link_2"/>
    <origin rpy="1.5707963267948966   0 3.141592653589793" xyz="0 0 0.2075"/>
    <axis xyz="0 0 1"/>
    <limit effort="320" lower="-2.0943951023931953" upper="2.0943951023931953" velocity="10"/>
  </joint>
  <link name="link_2">
    <inertial>
      <origin rpy="0 0 0" xyz="1.2e-05 0.06472 0.004804"/>
      <mass value="8.807"/>
      <inertia ixx="0.0824701683534692" ixy="0" ixz="0" iyy="0.0164110968055191" iyz="0" izz="0.0873510892661361"/>
    </inertial>
  </link>
  <!-- joint between link_2 and link_3 -->
  <joint name="joint_3" type="revolute">
    <parent link="link_2"/>
    <child link="link_3"/>
    <origin rpy="1.5707963267948966 0 3.141592653589793" xyz="0 0.2325 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="320" lower="-2.9670597283903604" upper="2.9670597283903604" velocity="10"/>
  </joint>
  <link name="link_3">
    <inertial>
      <origin rpy="0 0 0" xyz="-2.08e-04 0.01601 0.087283"/>
      <mass value="2.8"/>
      <inertia ixx="0.0230152941318817" ixy="0" ixz="0" iyy="0.0228661554143474" iyz="0" izz="0.0554482283748983"/>
    </inertial>
  </link>
  <!-- joint between link_3 and link_4 -->
  <joint name="joint_4" type="revolute">
    <parent link="link_3"/>
    <child link="link_4"/>
    <origin rpy="1.5707963267948966 0 0" xyz="0 0 0.1875"/>
    <axis xyz="0 0 1"/>
    <limit effort="320" lower="-2.0943951023931953" upper="2.0943951023931953" velocity="10"/>
  </joint>
  <link name="link_4">
    <inertial>
      <origin rpy="0 0 0" xyz="-2.03e-04 0.098639 0.011478"/>
      <mass value="5.283"/>
      <inertia ixx="0.0472789668341968" ixy="0" ixz="0" iyy="0.00964726804146909" iyz="0" izz="0.0466529007761679"/>
    </inertial>
  </link>
  <!-- joint between link_4 and link_5 -->
  <joint name="joint_5" type="revolute">
    <parent link="link_4"/>
    <child link="link_5"/>
    <origin rpy="-1.5707963267948966 3.141592653589793 0" xyz="0 0.2125 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="320" lower="-2.9670597283903604" upper="2.9670597283903604" velocity="10"/>
  </joint>
  <link name="link4_col_link"/>
  <joint name="link4_col" type="fixed">
    <origin rpy="0 0 0" xyz="0 0.3 0.0"/>
    <parent link="link_4"/>
    <child link="link4_col_link"/>
    <axis xyz="0 0 0"/>
  </joint>
  <link name="link_5">
    <inertial>
      <origin rpy="0 0 0" xyz="5.7e-05 0.014468 0.062938"/>
      <mass value="1.889"/>
      <inertia ixx="0.0138359953358589" ixy="0" ixz="0" iyy="0.0116859337751969" iyz="0" izz="0.00571863785412578"/>
    </inertial>
  </link>
  <!-- joint between link_5 and link_6 -->
  <joint name="joint_6" type="revolute">
    <parent link="link_5"/>
    <child link="link_6"/>
    <origin rpy="1.5707963267948966 0 0" xyz="0 0 0.1875"/>
    <axis xyz="0 0 1"/>
    <limit effort="320" lower="-2.0943951023931953" upper="2.0943951023931953" velocity="10"/>
  </joint>
  <link name="link_6">
    <inertial>
      <origin rpy="0 0 0" xyz="3.63e-04 0.016982 -0.019731"/>
      <mass value="2.32"/>
      <inertia ixx="0.00732758787216765" ixy="0" ixz="0" iyy="0.00477633755783711" iyz="0" izz="0.00606797638550296"/>
    </inertial>
  </link>
  <!-- joint between link_6 and link_7 -->
  <joint name="joint_7" type="revolute">
    <parent link="link_6"/>
    <child link="link_7"/>
    <origin rpy="-1.5707963267948966 3.141592653589793 0" xyz="0 0.0796 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="320" lower="-3.0543261909900763" upper="3.0543261909900763" velocity="10"/>
  </joint>
  <link name="link_7">
    <inertial>
      <origin rpy="0 0 0" xyz="0.001904 -9.3e-05 0.018862"/>
      <mass value="0.863"/>
      <inertia ixx="0.000884" ixy="0" ixz="0" iyy="0.000888" iyz="0" izz="0.001105"/>
    </inertial>
  </link>
  <!-- Link: Link7 - Endeffector -->
  <joint name="joint_ee" type="fixed">
    <parent link="link_7"/>
    <child link="end_effector_link"/>
    <origin rpy="0 -1.575 -1.575" xyz="0 0 0.21"/>
  </joint>
  <link name="end_effector_link">
    </link>
  <link name="end_effector_col_link"/>
  <joint name="end_effector_col" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.13"/>
    <parent link="link_7"/>
    <child link="end_effector_col_link"/>
    <axis xyz="0 0 0"/>
  </joint>
</robot>
